librz/arch/isa/msp430/msp430_il.c

```diff
@@ -0,0 +1,333 @@
+// SPDX-FileCopyrightText: 2024 Mostafa Mahmoud <ubermenchun@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <msp430/msp430_il.h>
+
+// ``The 16-bit program counter (PC/R0) points to the next instruction to be executed.``
+#define PC_SIZE 16
+// ``The low byte of a word is always an even address. The high byte is at the next odd address.
+//   For example, if a data word is located at address xxx4h, then the low byte of that data word
+//   is located at address xxx4h, and the high byte of that word is located at address xxx5h.``
+#define IS_BIG_ENDIAN false
+// implied by the width of the PC and other registers (which are used as pointers in the relevant addressing modes)
+#define MEM_ADDR_SIZE 16U
+
+#include <msp430/msp430_register_names.h>
+#include <msp430/msp430_il_getset.inc>
+#include <msp430/msp430_il_flags.inc>
+#include <msp430/msp430_il_jmp_utils.inc>
+
+#include "rz_il/rz_il_opbuilder_begin.h"
+
+// ************************************* ------------------- ********************************* //
```

Comment:
Instead of this I recommend Doxygen grouping, see how it's done in other files as well. See https://www.doxygen.nl/manual/grouping.html